### ISSUE TITLE
Nest.js Graphql typegeneration

### ISSUE THREAD

**Issue by aalpgiray** [nestjs/graphql#123]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behaviour
When type generation mode is class and if I chose to use property arguments the generated class is not abstract. Therefore, the typescript compiler fails to build because abstract methods can only be used in abstract classes.

``` typescript
@Module({
  imports: [
    AuthorsModule,
    GraphQLModule.forRoot({
      typePaths: ['./**/*.graphql'],
      definitions: {
        path: join(process.cwd(), 'src/graphql.ts'),
        outputAs: 'class',
      },
    }),
  ],
})
export class AppModule {}
```

``` graphql
type Author {
  id: Int!
  firstName: String
  lastName: String
  posts(id: Int, postName: String): [Post]
}
```

``` typescript
export class Author {
    id: number;
    firstName?: string;
    lastName?: string;

    abstract posts(id?: number, postName?: string): Post[] | Promise<Post[]>;
}
```

## Expected behaviour
The generated type should be abstract if there is no other implementation issue with that.

``` typescript
export abstract class Author {
    id: number;
    firstName?: string;
    lastName?: string;

    abstract posts(id?: number, postName?: string): Post[] | Promise<Post[]>;
}
```

## Environment

<pre><code>
Nest version: X.Y.Z

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by ave88** [comment]
For fast solution you can use 'interface' option.
```typescript
export interface Author {
    id: number;
    firstName?: string;
    lastName?: string;

    posts(id?: number, postName?: string): Post[] | Promise<Post[]>;
}
```

**Comment by ave88** [comment]
I have problem with "root" types
user.graphql file:
```graphql
type Query {
  users(filter: FilterInput ....): UsersResponse!
}

type Mutation {
  create(input: CreateInput!): CreateResponse!
}

type CreateResponse {
  data: User
  query: Query
}
```
But in generated graphql.schema.ts will have:
```typescript
export interface IQuery {
  users(filter: ...): ....
}

export interface CreateResponse {
    data?:User;
    query?: Query;
}
```
As you can see Query type generate into "IQuery" but in response type it's still "Query"

**Comment by kamilmysliwiec** [comment]
Should be fixed in 5.5.3
